### ISSUE TITLE
Creature OnBlocked event is not fired by placeables (and tileset)

### ISSUE THREAD

**Issue by Shad000w** [Beamdog/nwn-issues#116]
**To Reproduce**

 - Load attached save. [000880 - onblocked.zip](https://github.com/Beamdog/nwn-issues/files/4908752/000880.-.onblocked.zip)
 - Run around the blocked skeletons and observe the debug messages. Notice they are not saying they are blocked by Bench, but by other skeletons at best.

**Specifics**

 - OS: any
 - Game build: any
 - Distro platform: any
 - Content overrides? doesn't matter

**If needed, describe the bug**

Creature OnBlocked event fires only for doors and creatures. At least placeables should be easy to add into the conditions. It would be great to have the script fired when blocked by tileset (no path to target) but that is an extra.

**Comment by Finaldeath** [comment]
Not so much a bug (ie; Bioware never added this) but Bioware did add Creatures to the OnBlocked event eventually, placeables (at least bashable ones) might be also good to see from an AI perspective.

The default AI won't do anything about blocking creatures, or if they were added, placeables (it tests GetIsDoorActionPossible) so won't break anything I'd guess.

Would be nice however to really have OnBlocked fire for any kind of major pathfinding failure (if the pathfinder was substantially improved) since it is almost comically impossible to find out if a creature just can't reach another location (eg; they're on the other side of a river, or we're stuck slightly on a neutral creature, or we're too big to get through a doorway).
